### PR TITLE
Add `StringSection` node to the ASR

### PR DESCRIPTION
**Pull request by namannimmo10**

### CODE REVIEW

File: src/libasr/ASR.asdl [hunk 1]
```diff
@@ -247,6 +247,7 @@ expr
     | StringRepeat(expr left, expr right, ttype type, expr? value)
     | StringLen(expr arg, ttype type, expr? value)
     | StringItem(expr arg, expr idx, ttype type, expr? value)
+    | StringSection(expr arg, array_index section, ttype type, expr? value)
```
Comment (certik) [inline src/libasr/ASR.asdl:250]: I would probably do it like this:
```suggestion
    | StringSection(expr arg, expr start, expr end, expr step, ttype type, expr? value)
```
And `step` is always present, and equal to 1 if it is not specified in AST explicitly.

**Comment by certik** [comment]
This looks much cleaner! Is this ready for review?

**Comment by namannimmo10** [comment]
Yes, this is ready!

**Review by certik — APPROVED** [review]
I think that this is good, to merge. Thanks for doing this!

**Pull request merged**